### ISSUE TITLE
Open for collaboration

### ISSUE THREAD

**Issue by martinberlin** [martinberlin/CalEPD#1]
This project is open to any external collaboration. Coding/ testing or even ideas about architecture are always welcome.

This is are my current [Epaper displays](http://cale.es/eink-displays) for testing: 

- Waveshare 7.5 800x480 (Given to Carlos my partner in this project)
- Good display 5.83" | GDEW0583T7 b/w/r 600x448 (Only 3 Color Red/Black/White)
- Good display 4.2" | GDEW042T2 b/w 400x300
- Good display 2.13 b/w gdeh0213b73 (Used in TTGO)
- Good display 2.7 b/w gdew027w3 (TTGO T5S)

Update: New models are only going to appear on the mother project wiki, only after being tested will land here

https://github.com/martinberlin/cale-idf/wiki

Just leave your comments and ideas if you are interested to participate in this project.
Please add your test epapers in case you have any that is not on the list and you are interested on having the class added to CalEPD ESP-IDF Component.

**Comment by goran-mahovlic** [comment]
Hi, 

thank you for this library and all other work on open source projects! 
In few past day I am working on some repository that we could use for startup point of EEZ Studio LVGL project.

https://github.com/eez-open/studio

https://www.envox.eu/studio/studio-introduction/

Recent version of studio is now supporting LVGL projects (still in sper early stage) so I am trying to add all needed background for ESP32.
I have used some fork of esp32_lvgl that is using LVGL 8.2. then did some small fixes to get everything running on 8.3 that studio is currently using. 
I have checked and got some results with TFT screens, and then I noticed your library so I also added your e-ink repo.
I have GDEW042T2 and Inkplate 6 (that is using other lib).
If I set some text on screen inside your lib everything works! SO that is good start, but I cannot get it to run LVGL "things"...
I see you experimented much more with it so maybe you can give me some clue why I just get blank screen or sometimes some parts of screen.
I have checked with your demo example but I get similar (maybe little bit better results).

My work is here and it would be great if you could help me getting lvgl ink that would be great.: https://github.com/goran-mahovlic/esp32-lvgl-eez-demo

**Comment by martinberlin** [comment]
Lvgl is a complicated thing for epaper. I got it working only in version 7 so I have no clue what can happen in  . 8.x
One important point is that what it renders should be black if it’s a monochrome epaper. I must say that better results are on parallel displays driven by EPDiy project.
On the other side of things I lost a bit my interest because I made a PR that will never get merged due to License issues

**Comment by goran-mahovlic** [comment]
Ok, so I am on my own on this one :)

If you have at least bit of time maybe you can check under components if I am using good versions of your repos?

At some point I have noticed that issue really bad to trow away so much work. I am in contact with NLnet so maybe I could ask them to recheck issue and at least comment license problems or maybe find possible solution (I have send mail to them). It would help if you point me to a pull request so I do not need to search again (Found it!).

Anyhow studio can at some point use drivers directly - that is how it was used before LVGL, but I do not think that is on Envox priority list. But it would be great to have great GUI builder on those screens...

**Comment by martinberlin** [comment]
No I'm interested and will help whenever I can. But it turns me down that your work cannot be merged and hence remains somehow in the dark.
Keep me informed, you have also my email in my github profile and you can find me also in epdiy.slack.com (just tell me yours and I send you an invitation)

**Comment by goran-mahovlic** [comment]
Thank you!

 Yes I do not like that lvgl for esp32 did not move for like years :( And it is stupid for studio to add support for 7.9 today.
And yes main problem is that we will not able to help them as I see so much open pull requests, but I am working on a proof of concept and remote hiden repo is ok for now. 
If they update esp driver I will try to use updated version and try to help in testing.

My mail is: goran.mahovlic@gmail.com - I have slack but rarely using it. 

If you are using discord I am here: 

This is Envox: https://discord.gg/WtTVmfZmys

This is my Makerspace Radiona: https://discord.gg/QtvyHQUMp8

**Comment by hraftery** [comment]
I'm looking to add:

* Good Display 2.13 inch b/w/r gdey0213z98

I know your focus is on >= 400x300 px, but I hope this is a useful contribution to the space anyway.

I have the display working with the vendor-provided Arduino demo project. Any pointers on prior art or how best to get started are gratefully received.

**Comment by martinberlin** [comment]
Hi @hraftery sure please check existing classes and add your model. There is for sure one that gas a similar controller

**Comment by hraftery** [comment]
Sorry after some investigation I see that you need a port of Adafruit_GFX_Library anyway, which relies on big chunks of Arduino. Meanwhile ESP-IDF has `arduino-esp32` as a component, so adding Arduino native libraries is fairly well supported. I love the idea of a native EPD library, but for my immediate purposes pulling in the entire Arduino library is not the end of the world.

I've now got GxEPD2 up and running in ESP-IDF for this display, with a few untouched Arduino libraries as dependencies.

**Comment by martinberlin** [comment]
> Sorry after some investigation I see that you need a port of Adafruit_GFX_Library anyway, which relies on big chunks of Arduino. 

That's not correct. The only "big chunk of Arduino" it needs is the print function. So you don't need Arduino at all.
If you want to use Arduino IDE anyways you are in the wrong repository anyways @hraftery this is for Espressif IDF

**Comment by hraftery** [comment]
Nasty. Glad I found a different way.
